backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramImagePathService.java

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.spring.collaborative.diagrams;
+
+import java.util.List;
+
+import org.eclipse.sirius.web.api.services.IImagePathService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Register the diagram images folder.
+ *
+ * @author sbegaudeau
+ */
+@Service
+public class DiagramImagePathService implements IImagePathService {
```

Comment:
It's not clear to me what a "diagram image" is. An image used/usable by a shape on a diagram? An image (icon) to represent a specific diagram instance? Or a particular type of diagram (i.e. the same image for all instances of a given `DiagramDescription`)? Or an image to reprent the "diagram" kind of representation (as opposed for trees/forms/etc.)?

Reply:
It does not really matter here, this class is used to register a path where some images related to diagram representation can be stored. The role of those images is not very relevant, it can be anything. For example, we could return and use some SVG images in the NodeImageStyle with hardcoded image paths for images located in this folder.